app/views/components/_alert.html.erb

```diff
@@ -0,0 +1,19 @@
+<%# This is a simple flash component
+    It uses local_assigns.fetch to get variables.
+    
+    Example usage: (the % are removed from the < and >)
+    < render "components/alert", type:"success" do >
+        This is a success alert with <a href="#" class="alert-link">an example link</a>.
+    < end >
+
+    It has the following types:
+    alert,success,notice
+%>
+
+<% message = yield.presence || local_assigns.fetch(:message) %>
```

Comment:
A default message should be passed to `fetch` else we get an error when there is no message.